artifact/tar/tar_test.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <cstdint>
+#include <string>
+#include <vector>
+#include <fstream>
+#include <ostream>
+
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <common/processes.hpp>
+
+#include <artifact/tar/tar.hpp>
+
+using namespace std;
+
+namespace io = mender::common::io;
+
+namespace tar = mender::tar;
+
+namespace processes = mender::common::processes;
+
+class TarTestEnv : public testing::Test {
+protected:
+	static void SetUpTestSuite() {
+		string script = R"(#! /bin/sh
+
+# Create small tar file
+echo foobar > testdata
+tar cvf test.tar testdata
+
+# Create large tar file
+dd if=/dev/random of=testinput.large bs=1M count=4
+tar cvf test-large.tar testinput.large
+
+exit 0
+)";
+
+		const string script_fname = "./test-script.sh";
```

Comment:
Suggest using [`TemporaryDirectory`](https://github.com/mendersoftware/mender/blob/feature-c%2B%2B-client/common/testing.hpp#L26) and grab its `Path()` to put the things in a temporary directory. That both reduces the chance that you overwrite something in the current directory, and you can also remove most of your cleanup code, because it does it automatically. You will need to construct the object here, and destroy it in the cleanup code though.